backend/dataall/core/notifications/services/email_notification_provider.py

```diff
@@ -0,0 +1,77 @@
+# Email Notification Provider implements the email notification service abstract method
+import logging
+
+from dataall.base.aws.cognito import Cognito
+from dataall.base.aws.ses import Ses
+from dataall.base.config import config
+from dataall.core.notifications.services.email_notification_service import EmailNotificationService
+
+log = logging.getLogger(__name__)
+
+notification_configs = config.get_property('core.share_notifications')
+
+
+class EmailNotificationProvider(EmailNotificationService):
+
+    def __init__(self, dataset_owner_group, dataset_stewards, requester_group_name, requester_email_id, email_client) -> None:
+        super().__init__()
+        self.dataset_owner_group = dataset_owner_group
+        self.dataset_stewards = dataset_stewards
+        self.requester_group_name = requester_group_name
+        self.requester_email_id = requester_email_id
+        self.email_client = email_client
+
+    # Implementation
+    def send_email(self, to, message, subject):
+        self.email_client.send_email(to, message, subject)
+
+    @staticmethod
+    def get_email_ids_from_groupList(group_list, identity_provider):
+        email_list = set()
+        for group in group_list:
+            email_list.update(identity_provider.get_user_emailids_from_group(group))
+        return email_list
+
+    @staticmethod
+    def get_email_provider_instance(datasetOwnerGroup, datasetStewards, requesterGroupName, requesterEmailId):
+        return EmailNotificationProvider(datasetOwnerGroup, datasetStewards, requesterGroupName, requesterEmailId, Ses.get_ses_client())
+
+    @staticmethod
+    def send_email_task(requesterEmailId, subject, message, data):
+        # Get instance of the email provider
+        email_provider = EmailNotificationProvider.get_email_provider_instance(
```

Comment:
The data dict is very specific to share notifications. Can we make it more generic? Like data.get('senderGroups'). data.get('recipientGroup')

Reply:
We can do it in this PR or in the one that I am working on to move notifications to modules as a standalone module that does not depend on datasets. What do you think?

Reply:
As discussed, I have handled and refactored such that it becomes generic. Now there will will be two fields i.e. recipient_group_list and recipient_email_list ( in case group_notifications config is false this variable will contain the requester email id ) . These fields will be instead of the current fields like datasetOwnerGroup, datasetStewardsGroup, requesterGroupName

Reply:
Awesome!!